### ISSUE TITLE
Typescript powerMonitor Issues

### ISSUE THREAD

**Issue by vsvipul** [electron/electron#18966]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details
Sometimes, developers might import electron into a file using statements like these-
```
import electron from 'electron';
```
or
```
import * as electron from 'electron';
```
instead of 
```
import electron = require('electron');
```
The first two statements sometimes cause problems in Typescript and give errors on running the app-
```
[start-app] App threw an error during load
[start-app] Error: Cannot require "powerMonitor" module before app is ready
[start-app]     at binding (internal/bootstrap/loaders.js:82:36)
[start-app]     at process.atomBinding (/home/vipul/Documents/zulip-desktop/node_modules/electron/dist/resources/electron.asar/common/atom-binding-setup.js:4:14)
[start-app]     at Object.<anonymous> (/home/vipul/Documents/zulip-desktop/node_modules/electron/dist/resources/electron.asar/browser/api/power-monitor.js:2:46)
[start-app]     at Object.<anonymous> (/home/vipul/Documents/zulip-desktop/node_modules/electron/dist/resources/electron.asar/browser/api/power-monitor.js:25:3)
[start-app]     at Module._compile (internal/modules/cjs/loader.js:711:30)
[start-app]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
[start-app]     at Module.load (internal/modules/cjs/loader.js:620:32)
[start-app]     at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
[start-app]     at Function.Module._load (internal/modules/cjs/loader.js:551:3)
[start-app]     at Module.require (internal/modules/cjs/loader.js:658:17)
[start-app] A JavaScript error occurred in the main process
```
This is particularly related to powerMonitor being loaded when electron is imported. 

* **Electron Version:**
  * v3.1.10
* **Operating System:**
  * Ubuntu 19.10
* **Last Known Working Electron version:**
  * NA

### Expected Behavior

Successful import without any errors while running.

### Actual Behavior

Importing electron gives the error-
```
Error: Cannot require "powerMonitor" module before app is ready
```

### To Reproduce

Using-
```
import * as electron from 'electron';
```

**Comment by MarshallOfSound** [comment]
Can you provide a minimal ts repro that does this.  I use that import statement in multiple ts apps and do not have this issue

**Comment by magicdawn** [comment]
for anyone encounter this. 

this is caused by esModuleInterop + importStar `import * as electron from 'electron'`

```js
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
```

__importStar will try to get `electron.powerMonitor` when import.
so the solution is avoid import * when you set `esModuleInterop` flag to true

**Comment by nornagon** [comment]
This is fixed in v10.0.0, powerMonitor can now be required before app ready (though it still can't be used before app ready).
